### DISCUSSION TITLE
OAuth2.0 & Mirth 3.7

### DISCUSSION THREAD

**Discussion by ggVert** [nextgenhealthcare/connect#4775] [Q&A]
Hello,
I need some help to use OAuth2.
I have to use OAuth2.0 with flow Client Credentials with mirth 3.7 (mandatory) 
I create a HTTP Sender to the site who delivers the token
I do a POST with the clientId as username  and the client_secret as password.
In parameters I put grant_type = client_credentials and the scope.
In the Headers I put the Autorization whith the basic authentication., "application/x-www-form-urlencoded" in current Type.
It seems to work i have  a status to 200 in response but no data in the message.
Did I miss something?
Thanks for your help
![mirth3 7](https://user-images.githubusercontent.com/92116712/136440723-d9c4ec10-ea09-481a-91e2-afb6ae609421.png)
.

**Comment by pacmano1** [comment]
Not an answer to question, just a note:  If you are on 3.7.0, please see https://github.com/nextgenhealthcare/connect/issues/4239

**Comment by michaeljeffreycollins** [comment]
you have a username and pass but then have a header for authorization as well? I would double check the headers to make sure that is needed with user and password auth to oAuth 2.  

Also you don't list a client_assertion or client_assertion_type in the parameters.
